### DISCUSSION TITLE
Issue with WiFi AP

### DISCUSSION THREAD

**Discussion by orbitcoms** [Mair/esp32-course#162] [Course Qeustions] [answered]
I have followed along the Wifi Connection videos for connection as both AP and then STA.
The STA connects ok asnd gets IP but I have an issue with the AP.

Below in my code I have set the ssid to "My_ESP" and pw tyo "1234".
When I run the code I get "AP Started" message from vent handler ok but when I browse my wifi network the AP comes up wit6h wrong SSID and will not accept my pw. The SSID I get when it runs is "ESP_7C141D" insrtead of "MJy_ESP".

I also get a wiggly line warning in both wifi_connect routines under "ESP_IF_WIFI_AP" and again in "ESP_IF_)WIFI_STA" in the line esp_wifi_set_config

The error states Warning: implicit conversion from enumeration type "esp_interface_t" to nfiofferent enumeration type "wifi_interface_t".

Here is my main calling task code

void wifi_connect(void * params)
{
	wifi_connect_ap("My_ESP", "1234");
	
	for (size_t i = 60; i > 0; i--)
	{
		printf("Disconnecting AP in %d\n", i);
		vTaskDelay(pdMS_TO_TICKS(1000));
	}
	wifi_disconnect();
......
}

And the connect code for AP

void wifi_connect_ap(const char *ssid, const char *pass)
{
	esp_netif = esp_netif_create_default_wifi_ap();
	
	wifi_config_t wifi_config;
	memset(&wifi_config, 0, sizeof(wifi_config_t));
	strncpy((char *)wifi_config.ap.ssid, ssid, sizeof(wifi_config.ap.ssid) - 1);
	strncpy((char *)wifi_config.ap.password, pass, sizeof(wifi_config.ap.password) - 1);
	//wifi_config.ap.authmode = WIFI_AUTH_OPEN;
	wifi_config.ap.authmode = WIFI_AUTH_WPA_WPA2_PSK; //use this if password protected
	wifi_config.ap.max_connection = 4;
	esp_wifi_set_mode(WIFI_MODE_AP);
	esp_wifi_set_config(ESP_IF_WIFI_AP, &wifi_config);
	esp_wifi_start();
}

**Comment by Mair** [comment]
Hi,
no idea why your SSID is incorrect,
maybe print out the SSID just as you enter the function to ensure it is what you think it is.
However, the minimum password for a WIFI device is 8 chars

**Comment by orbitcoms** [comment, marked as answer]
That was it. my pw was less than 8 chars.
